src/Scope.h

```diff
@@ -37,10 +37,22 @@ class Scope : public Obj
 
 	const IDPtr& Find(std::string_view name) const;
 
-	template <typename N, typename I> void Insert(N&& name, I&& id)
+	void Insert(std::string name, IDPtr id)
 		{
-		local[std::forward<N>(name)] = std::forward<I>(id);
-		ordered_vars.push_back(std::forward<I>(id));
+		local[name] = id;
+		ordered_vars.push_back(id);
+		}
+
+	void RemoveGlobal(std::string name, IDPtr /* gid */)
+		{
+		local.erase(name);
+
+		// We could remove the identifier from ordered_vars, but for
+		// now we skip doing so because (1) the only removals we do are
+		// for global scope (per the method name), and the only use
```

Comment:
The intention behind the naming was to flag that it won't work properly if applied to a non-local scope (because it doesn't remove the element from `ordered_vars`).  I'm not sure what you mean about delegating the call to `global_scope()`, can you sketch that?  In any case, I agree that there should be a check/assert that the method is only used for `global_scope()`.

Reply:
> I'm not sure what you mean about delegating the call to global_scope(), can you sketch that?

I don't think it would look any better. Maybe a `zeek::detail::remove_global()`, but would still need some accessor/friending on `Scope` to actually do the removal.

This is `zeek::detail`, so less friction for improvements/renaming is low.